### PR TITLE
Merge feature/105-template

### LINKED ISSUE #105: Allow another template file to be used for vhost:create
Add `--from-file` flag to `vhost:create` to use external files as template.

### PR DESCRIPTION
**Pull request by stevenrombauts**
Closes #105

### CODE REVIEW

**Comment by stevenrombauts** [comment]
@yiendos Would you mind giving this a quick test please? Here are the steps:

1. Create a site `joomla site:create foo`. Make sure it still works.
1. Create a sample config file: 

    ``` 
    echo "<VirtualHost *:80>
      ServerName %site%.test
      RewriteEngine on
      RewriteRule ^ http://joomla.box [R=301,L]
    </VirtualHost>" > /tmp/sample.conf
    ```

1. Use this file as template for the Apache config: 

    ```
    joomla vhost:create foo --apache-template=/tmp/sample.conf
    ```

File: src/Joomlatools/Console/Command/Vhost/Create.php [hunk 1]
```diff
@@ -144,4 +127,66 @@ protected function execute(InputInterface $input, OutputInterface $output)
             @unlink($tmp);
         }
     }
+
+    protected function _getVariables(InputInterface $input)
+    {
+        $documentroot = Util::isPlatform($this->target_dir) ? $this->target_dir . '/web/' : $this->target_dir;
+
+        $variables = array(
```
Comment (yiendos) [inline src/Joomlatools/Console/Command/Vhost/Create.php]: @stevenrombauts genius move here, naming the variables here in such a manner makes the template files so much more readable.

**Comment by yiendos** [comment]
@stevenrombauts looking good, I didn't test more exotic flavours such as:
https://github.com/joomlatools/joomlatools-console/pull/106/files#diff-935be7104705247f733d1207e5427a7cR166

Did you want me to test?

**Comment by stevenrombauts** [comment]
@yiendos Thanks! Don't worry about the kodekit versions, that hasn't even been completed in #79, I'll wrap it up there before releasing next version.

**Pull request merged**